### PR TITLE
Fix missing semicolons

### PR DESCRIPTION
**Pull request by AnsonYeung**
Add semicolons in several places in the codegen

This fixes JS codegen on the IR `Label(lbl, Break(lbl, true), End())`

### CODE REVIEW

**Comment by LPTK** [comment]
Thanks! But there is no need to add semicolons after things like function definitions and classes. `class Foo {} console.log("ok")` is valid JS.

BTW, the CI fails due to flakiness for a reason @chengluyu has already fixed in a separate PR.

File: hkmc2/shared/src/main/scala/hkmc2/codegen/js/JSBuilder.scala [hunk 1]
```diff
@@ -208,19 +208,19 @@ class JSBuilder extends CodeBuilder:
...
               case N =>
                 fun match
-                case S(f) => doc"${f}; # ${sym.nme}.class = ${clsJS}"
+                case S(f) => doc"${f}; # ${sym.nme}.class = ${clsJS};"
                 case N => clsJS
         thisProxy match
-          case S(proxy) => doc" # const $proxy = this; # ${res.stripBreaks};${returningTerm(rst)}"
-          case N => doc"$res;${returningTerm(rst)}"
+          case S(proxy) => doc" # const $proxy = this; # ${res.stripBreaks}${returningTerm(rst)}"
+          case N => doc"$res${returningTerm(rst)}"
```
Comment (LPTK) [inline hkmc2/shared/src/main/scala/hkmc2/codegen/js/JSBuilder.scala:220]: What's the rationale for removing semicolons in two places and adding them back in five places instead?

Comment (AnsonYeung) [reply hkmc2/shared/src/main/scala/hkmc2/codegen/js/JSBuilder.scala:220]: This is to remove the semicolon for class and functions

Comment (LPTK) [reply hkmc2/shared/src/main/scala/hkmc2/codegen/js/JSBuilder.scala:220]: Ah I see it's just to avoid adding one for the function case.

**Pull request merged**